### DISCUSSION TITLE
Dubbio Caddy per configurazione multisiti + HA

### DISCUSSION THREAD

**Discussion by labattazza** [indomus/forum#1287] [Networking]
Buondì a tutti, chiedo un paio di chiarimenti perché sto valutando di rifare la mia installazione di HA, partendo come prima cosa dall'installazione di Caddy.
 
 Nella mia rete casalinga ho due server: quello di un amico, su cui c'è installato Nextcloud, raggiungibile attraverso redirect della porta 11111 sulla 443, e il mio  NUC, sui cui tra gli altri girano un server apache + nextcloud + HA.
 
 Su entrambi i server uso certificati SSL autofirmati.
 
 Uno dei miei dubbi è questo:  è possibile configurare il reverse proxy in modo che col telefono posso accedere ai servizi del NUC senza passare dalla WAN, quando sono all'interno della LAN? 
 Cioè se punto a NUC.miosito.net dall'interno della LAN e per qualche motivo mi è caduta la connessione internet (es: un'auto si è schiantata sull'armadio fibra ottica sotto casa) posso comunque raggiungere nextcloud/HA?
 Installando i certificati di letsecrypt tramite Caddy non dovrei più preoccuparmi dell'avviso di sicurezza che ogni volta compare per i certificati autofirmati, e lasciar perdere la tediosità di configurarli su apache su ogni server, corretto?
 
 Altro dubbio: HA lo raggiungo tramite nome dominio ottenuto da duckdns, con l'addon configurato. Purtroppo il proxy aziendale blocca il traffico verso i domini duckdns, per cui accedo solo tramite telefono. 
 Per accedere a nextcloud uso il nome dominio ottenuto con NOIP, le cui credenziali sono inserite nel fritz che supporta nativamente. A questo punto potrei usare solo il nome dominio di NOIP, semplificando tutto, magari configurando CLOUD.dominioNOIP.net su 10.0.0.50:80 per raggiungere nextcloud e HA.dominioNOIP.net su 10.0.0.50:8123 per Home Assistant. E magari AMICO.dominio.NOIP.net su 10.0.0.80:80 pe il server del mio amico.
 Ma se volessi mantenere i due nomi dominio, entrambi col il proprio certificato SSL, sarebbe comunque possibile?
 
 Grazie dei chiarimenti!

**Comment by snakuzzo** [comment]
Per quanto riguarda i domini, puoi usarne anche N (quindi duckdns e noip), non c'è alcun problema.

Es. in `Caddyfile`

```
https://miodominio.duckdns.org:9000 {
	reverse_proxy 192.168.1.10:8123
}
https://miodominio.ddns.net:8400 {
	reverse_proxy 192.168.1.10:8123
}
```

Quanto al discorso da rete interna, se non puoi configurare dei filtri direttamente sul tuo modem/router dovresti utilizzare qualcosa tipo pihole o adguard. Io utilizzo adguard che mi fa da server DNS (al posto di quello impostato sul router)...e lì puoi utilizzare dei filtri custom.
Es.

![adguard](https://user-images.githubusercontent.com/7205371/148904488-3c7fea93-2e3c-4b30-a2c3-ec585a121cd2.png)


Quindi quando provo a risolvere da un client sulla LAN che utilizza adguard come DNS...

![adguard1](https://user-images.githubusercontent.com/7205371/148904552-6f023b9f-590f-4535-81fd-ab81aa3f8e7b.png)

**Reply by labattazza** [reply]
cavolo, avevo risposto anche a questo valido suggerimento, ma non devo aver premuto su "invia"..

Comunque ora sto provando a installare pihole su una Raspberry che mi avanzava in casa, così elimino il problema delle pubblicità. E dopo vedo se riesco anche a far funzionare Caddy su Docker, così uso la Raspy come black box indipendente per tenere in ordine. 
Devo ravanare sulla configurazione del Fritz per capire se posso settare delle regole per la questione LAN/WAN, ma se ho capito bene usando la Raspy come server DNS con PiHole posso ottenere lo stesso risultato.

Mi metto al lavoro e vedo cosa riesco a ottenere.
Thanks!

**Reply by labattazza** [reply]
sono riuscito a configurare Docker sul Rasperry su cui ho installato PiHole (temporaneamente inattivo, per fare debug durante l'installazione di Caddy. Preferisco avere una variabile alla volta da tenere sotto controllo).

Ora che mi sono un po' impratichito con Docker Compose e sono riuscito a far reindirizzare sui due server casalinghi i dati in ingresso (nextcloud su un server, e un'altra installazione di nextcloud sul server che ospita anche Home Assistant e funge da mediacenter), vorrei ripartire da zero con Home Assistant Core su Docker.

Siccome in passato **/var** era cresciuta troppo per colpa di log fuori controllo e amenità del genere, per evitare che mi bloccasse il sistema l'ho spostata su una partizione da 20 giga.
Mi confermate che in una **installazione standard di Docker** le immagini e poi i container sono salvati sotto /var?

Ho visto anche come montare i volumi su altre cartelle, cosa molto utile.

Il comando **'docker-compose' se avviato in manuale** va dato nella cartella in cui c'è il file di configurazione .yaml. Ma se volessi far si che ad ogni avvio del server lui **parta in automatico** avviando i vari container che ho definito?

Un pezzo alla volta ce la posso fare :-D

**Reply by snakuzzo** [reply]
> Mi confermate che in una installazione standard di Docker le immagini e poi i container sono salvati sotto /var?

sì

> Ho visto anche come montare i volumi su altre cartelle, cosa molto utile.

che intendi dire?...A cosa ti riferisci di preciso ?

> Il comando 'docker-compose' se avviato in manuale va dato nella cartella in cui c'è il file di configurazione .yaml.

Non c'è alcuna regola. Di default se non specifichi nulla viene cercato il `docker-compose.yml` (o `.yml`) all'interno della stessa directory, ma si può specificare un qualsiasi file in qualsiasi altra directory con l'opzione `-f`. (per queste cose ti basta guardare l'help con `docker-compose --help`)

> Ma se volessi far si che ad ogni avvio del server lui parta in automatico avviando i vari container che ho definito?

L'autostart dei container Docker (che siano stati creati con il comando `docker` o con il tool `docker-compose`) lo stabilisci con il parametro restart. Leggi [QUI](https://docs.docker.com/config/containers/start-containers-automatically/)

**Reply by labattazza** [reply]
Grazie mille dei chiarimenti, non mi era chiaro il comportamento di "restart", mi studio un po' la documentazione, prima di fare altre domande.

Per la domanda sui volumi intendevo la possibilità di scambiare file tra il container e la macchina  che lo ospita visto che è un ambiente "confinato". Ad esempio in HA, se voglio guardare un film o ascoltare una canzone che sono in una cartella del sistema hosting.

**Reply by snakuzzo** [reply]
Ok...ma (non per fare il puntiglioso) nel precedente messaggio ti sei espresso male :)

**Reply by labattazza** [reply]
Macché puntiglioso! 
Mi ero espresso malissimo io, chiedo venia! Non sempre riesco a esser lineare nell'esporre il concetto, specie se sono molto stanco :-D

**Reply by snakuzzo** [reply]
ma vaaa tranquillo...si scherzava :)

**Comment by Scialla** [comment]
anche a me, dall'ufficio, il proxy blocca sia duckdns che noip MA, se indirizzo a una porta specifica, mi fa collegare.
Per capirci:
pippo.duckdns.org viene bloccato dal proxy
pippo.duckdns.org:7654 non viene bloccato dal proxy

di conseguenza ho configurato caddy in entrambi i modi

...prova

**Reply by labattazza** [reply]
ottima idea, purtroppo non c'è nulla da fare. Ho già provato più volte, specificando la porta anche ma nulla. il traffico di chrome e edge sul pc aziendale è filtrato, e visto che si tratta di strumento di lavoro non voglio rischiare a usare nulla che non sia ufficiale o rischio una lettera di richiamo, nel migliore dei casi.

**Comment by labattazza** [comment]
vi chiedo un consiglio "di rete" perché ho qualche difficoltà a far funzionare il tutto correttamente.

Per semplificare ho creato due schemi di come era la mia LAN casalinga, e di come è ora.

Prima tramite port forwarding inoltravo il traffico su due diversi server, entrambi col proprio certificato autofirmato.
Per non cambiare numero di porta ad Apache sul secondo server, avevo lasciato le classiche 80 e 443,  reindirizzando la 11200.
Funzionava tutto (avvertimento a parte da parte dei browser circa il potenziale rischio di sicurezza).
![Diapositiva1](https://user-images.githubusercontent.com/44818319/149528182-e9bca06f-89e5-45eb-8a10-c4df0ce27afc.JPG)

Ora ho deciso di centralizzare, installato Caddy, come da guida.
Lo schema è qua sotto.
La mia intenzione è avere un solo certificato per il dominio del tipo "casamia.ddns.net", per i vari server che stanno dietro. E reindirizzare il traffico in base alla porta che scelgo (o mi conviene provare qualcosa del tipo "nuc.casamia.ddns.net" e "mv.casamia.ddns.net"?)
Il problema è doppio:

- il proxy aziendale mi permette di collegarmi al server di casa mia sulla porta 80/443, ma non sulle altre. Per cui... no HA su porta standard e così via. 
	Se mi collego tamite browser cellulare il problema non c'è, e accedo ai file. 
	Se mi collego su pc aziendale a https:casamia.ddns.net ottengo la risposta del server "Saluti da casamia".
- dall'interno della LAN volevo connettere  il client nextcloud, ma la procedura di accesso tramite token non va a buon fine. Provando tramite browser ottengo un messaggio del tipo "il server genera una risposta http su una porta https" o simile. 
Questo perché se lasciavo attivo il sito "default-ssl" che c'è in /servers-available" sotto apache ottenevo errore: ipotizzo per il fatto che ci fossero due certificati SSL diversi in giro?
![Diapositiva2](https://user-images.githubusercontent.com/44818319/149528183-18dc7847-ea36-4c4d-9864-44e7a357c37f.JPG)

Qui sotto il file di configurazione di Caddy.
Dite che dovrei cambiare qualcosa "concettualmente" nella mia configurazione?
Non sono espertissimo di rete, per cui probabilmente sto affrontando la configurazione dal lato sbagliato.

Se ho ben compreso devo aprire la 80 o la 443 a Caddy per permettere il rilascio del certificato SSL da parte di Letsencrypt, quindi non posso raggiungere i server dietro il proxy su quelle porte. Per quel motivo sto provando con due porte alternative.
c'è sicuramente un'altra soluzione, può esser il caso di "nuc.casamia.ddns.net"? E dovrei modificare qualcosa sul server apache del NUC?

Spero di non esser stato troppo confusionario!
(https_header) {
```
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}

https://casamia.ddns.net { 
  respond "Saluti da casamia"
  file_server
}
https://casamia.ddns.net:11200 { 
  import https_header
  reverse_proxy http://IP_MV:80
}
https://casamia.ddns.net:11100 { 
  import https_header
  reverse_proxy http://IP_NUC:80
}
https://casamia.ddns.net:8123 { 
  import https_header
  reverse_proxy http://IP_NUC:8123
}
```

**Comment by labattazza** [comment]
Ciao, riprendo questo "topic" sperando che qualcuno più esperto di me in questioni di rete possa darmi una mano a scrivere il Caddyfile per farlo funzionae.

Dopo aver chiesto un consulto ai creatori di Caddy ho compreso che se scegliessi l'opzione "wildcard", cioè link di accesso del tipo `https://mv.labahouse.ddns.net` `https://nuc.labahouse.ddns.net` e così via sarebbe un po' complicata perché dovrei crearmi un container inserendo un plugin per i DNS di noip, che però al momento non esiste.

Allora la soluzione sarebbe quella delle "subfolder", che non è l'ideale ma fattibile.
Per cui link del tipo `https://labahouse.ddns.net/mv`

Il problema è come scrivere la regola.
Da quanto ho capito devo usare sia la direttiva `handle` che `rewrite` per riscrivere i link in modo che siano interpretabili dal server interno.

Qui sotto l'esempio del mio Caddyfile.
Anche provando ad accedere a una semplice pagina html sul server interno `10.0.0.40/test.htm`l ottengo un errore "Bad Request" (il server non sa interpretare la richiesta).

Qualcuno ha idee sull'errore che sto facendo? Sicuramente qualcosa di banale, purtroppo nion sono molto edotto su TCP/IP e annessi e connessi :-( e anche se sto cercando di formarmi, ho qualche bel problemino.

Grazie!

```
(https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}


https://labahouse.ddns.net {
        handle /mv* {
#                import https_header
                rewrite /mv* *
                reverse_proxy 10.0.0.40:80
        }

        handle {
                respond "Saluti da casamia"
        }
}
```

**Reply by Scialla** [reply]
io ci ho provato e riprovato e le subfolder non sono riuscito a farle funzionare e ho paura che non si possa fare.

Posso dirti però che ddns non consente l'utilizzo dei domini di 4 livello (`https://mv.labahouse.ddns.net`) e ci sei arrivato anche tu, ma duckdns invece si.

Io uso sia noip che duckdns con caddy e, con duckdns, uso tranquillamente i domini di 4 livello (ha.casamia.duckdns.org) mentre con noip non funziona

**Reply by labattazza** [reply]
nu, che brutta notizia!!
Mi pare che anche DynDns sia nella stessa situazione di Noip, giusto?

Il mio problema è che il proxy aziendale mi sega i domini sotto Duckdns: ho parlato col responsabile IT e mi ha detto che probabilmente il ranking di Duckdns è basso per colpa di qualche dominio che spara spam e malware. 
E' questo il motivo per cui ho usato Noip.

Se riesco a far funzionare il reverse con DuckDns posso accedere con qualsiasi dispositivo, tranne quello che uso di più durante il giorno e che mi serve per accedere i dati di casa. 

Cerco un servizio compatibile con Caddy e dimini di 3° livello.

Anzi, ora che ci penso, ho un dominio Aruba che uso come prove e spazio di archiviazione. Potrei reindirizzare quello sull'ip della mia linea, ma ora che non ho più Fastweb con l'ip fisso dovrei trovare un'altra soluzione, ma con un hosting condiviso le possibilità sono poche (un paio di anni fa forse avevo trovato su un blog una soluzione composta da un agent su raspberry e uno script php sul sito, ma non mi sento sicuro...)

Intanto grazie mille per avermi dato qualche punto fermo!

**Reply by snakuzzo** [reply]
continuo a non capire per quale motivo si parla di "compatibilità con caddy" ...

**Reply by labattazza** [reply]
Mi sono espresso male io (e non è la prima volta ;-) )

Intendo dire che sto cercando un fornitore di servizio dynamic DNS che abbia un modulo  già esistente per Caddy.

Come mi ha confermato "francislavoie" 
`I don’t know if there’s a noip DNS plugin for Caddy. But if there is, you’d need to build the docker image yourself by using a Dockerfile.`

e si riferisce a uno dei vari [moduli](https://caddyserver.com/docs/modules/) a disposizione.

**Reply by snakuzzo** [reply]
quindi ? Fallo col Dockerfile... io l'ho fatto per usare la dns-challenge per cloudflare

**Reply by snakuzzo** [reply]
Ma perché ti serve? Quello ti è utile soprattutto se non puoi esporre Caddy per la challenge HTTP (su porta 80) e per la challenge tls-apln (su porta 443) e quindi sei costretto a fare la dns-challenge.
no-ip ti permette di farla, ma dovresti pagare il servizio...con quello gratuito non puoi. Poi, oltre a questo, devi compilare Caddy con il modulo che ti permetta di farlo.

**Reply by snakuzzo** [reply]
Comunque...se vuoi cercare e provare...io per cloudflare avevo fatto così...

```yaml
  caddy:
    hostname: caddy
    container_name: caddy
    build:
      context: "./caddy/dockerfile"
      dockerfile: Dockerfile
    restart: always
    ...
    ...
```

Questo il `Dockerfile`

```
FROM caddy:builder AS builder
RUN xcaddy build --with github.com/caddy-dns/cloudflare
FROM caddy:latest
COPY --from=builder /usr/bin/caddy /usr/bin/caddy
```

e poi nel Caddyfile devi vedere come impostare la configurazione per farla (dipende poi dal modulo che usi)

```
https://homeassistant.prova.org:9000 {
	import https_header
	reverse_proxy http://192.168.1.10:8123
	tls miamail@gmail.com {
		dns cloudflare xxxxxxxxxxxxxxxxxx
	}
}
```

... dove `xxxxxxxxxx` sarebbe l'API Token di Cloudflare

```

**Reply by labattazza** [reply]
> quindi ? Fallo col Dockerfile... io l'ho fatto per usare la dns-challenge per cloudflare

è che da poco mi sono accostato al "mondo docker" e devo impratichirmi, non volevo iniziare a girare in tondo

> Ma perché ti serve? Quello ti è utile soprattutto se non puoi esporre Caddy per la challenge HTTP (su porta 80) e per la challenge tls-apln (su porta 443) e quindi sei costretto a fare la dns-challenge. no-ip ti permette di farla, ma dovresti pagare il servizio...con quello gratuito non puoi. Poi, oltre a questo, devi compilare Caddy con il modulo che ti permetta di farlo.

su Noip avevo già preso il servizio a pagamento (con l'amico che ha messo il suo server da me, visto che vive in campagna con unamisera Eolo a disposizione) per non dover rinnovare ogni mese. Ma anche quando usavo il free, non ho avuto problemi per il certificato SSL

> Comunque...se vuoi cercare e provare...io per cloudflare avevo fatto così...
> 

> e poi nel Caddyfile devi vedere come impostare la configurazione per farla (dipende poi dal modulo che usi)
> 
> ```
> https://homeassistant.prova.org:9000 {
> 	import https_header
> 	reverse_proxy http://192.168.1.10:8123
> 	tls miamail@gmail.com {
> 		dns cloudflare xxxxxxxxxxxxxxxxxx
> 	}
> }
> ```
> 
> ... dove `xxxxxxxxxx` sarebbe l'API Token di Cloudflare

a questo punto, visto che a te funziona, mi orientarei su questa soluzione.
Per cui se ho ben capito:

1. Creo il container seguendo le istruzioni che mi hai illustrato
2. Visto che ho già un nome dominio mio su Aruba, userei quello (https://www.labahouse.it)
3. Apro un account su Cloudfare
4. Setto sul router i(o sul raspberry) i parametri per il dynamic dns di Cloudfare
3. Sul pannello DNS di Aruba inserisco il nameserver di Cloudfare
4. nel caddyfile creo le impostazioni per `https://nuc.labahouse.it` `https://mv.labahouse.it`  `https://ha.labahouse.it` e cos' via

Concettualmente fila?

Grazie per gli esempi dettagliati

**Reply by snakuzzo** [reply]
- Devi spostare il dominio su cloudflare se vuoi fare così.
- Se il tuo router ti permette di aggiornare l'associazione ip-fqdn meglio, ma comunque sappi che c'è già qualcosa di pronto. Io uso un container docker creato con [questa immagine](https://hub.docker.com/r/oznu/cloudflare-ddns/).
- Non so cosa sia questo pannello DNS di Aruba
- Io ti ho illustrato come ho fatto...prendi spunto, non fare copia-incolla

**Reply by labattazza** [reply]
Grazie mille.
Appena ho un paio d'ore libere mi cimento!

> _* Devi spostare il dominio su cloudflare se vuoi fare così._
Al momento sono riuscito a mettere i DNS di Cloudfare nel pannello Aruba: non è sufficiente per usare il CNAME XX.labahouse.it? 
> _* Se il tuo router ti permette di aggiornare l'associazione ip-fqdn meglio, ma comunque sappi che c'è già qualcosa di pronto. Io uso un container docker creato con [questa immagine](https://hub.docker.com/r/oznu/cloudflare-ddns/)._
Ho trovato un paio di blog in cui descrivono i parametri di Cloudfare da inserire nella sezione DynDNS dei router per avere l'aggiornamento automatico. Il FritzBox permette anche configurazioni custom, ho creato l'API su Cloudfare, devo fare le prove. Se riesco a far funzionare sul router meglio, così sono indipendente da eventuali malfunzionamenti del Raspi
> _* Non so cosa sia questo pannello DNS di Aruba_
In Gestione Domini di Aruba c'è un pannello in cui si possono configurare i record DNS e inserire anche Nameserver esteni
> _* Io ti ho illustrato come ho fatto...prendi spunto, non fare copia-incolla_
Si, certo, lo adatto alle mie esigenze. Ad esempio credo di creare una cartella `caddy/dockerfile_cloudfare` in modo da tenermi configurazioni separate e tornare indietro facilmente se qualcosa non funziona fin da subito, così almeno riesco ad accedere attraverso quella che funziona, con il reindirizzamento porte.

Se ho capito bene in codice qua sotto nel Dockerfile crea un container caddy con l'apposito plugin cloudfare, al posto di quello "liscio". 
Però prima devo eliminare il container caddy che avevo creato prima, o scegliere un nome diverso
```
FROM caddy:builder AS builder
RUN xcaddy build --with github.com/caddy-dns/cloudflare
FROM caddy:latest
COPY --from=builder /usr/bin/caddy /usr/bin/caddy
```

Mi cimento, sperro proprio che il prossimo messaggio sarà "solved"!
Intanto grazie per gli spunti, così "studio" un po' meglio.

**Reply by labattazza** [reply]
> Io uso sia noip che duckdns con caddy e, con duckdns, uso tranquillamente i domini di 4 livello (ha.casamia.duckdns.org) mentre con noip non funziona

Ciao Scialla, giusto per curiosità: che soluzione hai adottato con noip? reindirizzamento delle porte?
Grazie!

**Reply by Scialla** [reply]
Sia reindirizzamento che "liscio" sulla 443

**Reply by labattazza** [reply]
Ciao, dopo lunga pausa dovuta a impegni vari che non lasciavano tempo per mettermi al pc con calma e testa libera, nei giorni scorsi sono riuscito, grazie ai preziosi consigli e a un po' di letture della documentazione su docker, a fare una installazione pulita su un secondo Raspberry per poter sperimentare Cloudflare senza interrompere i servizi attualmente attivi con noip.

Sono riuscito a creare l'immagine che comprende il plugin Cloudflare, e avviarla.
Seguendo la guida sul sito, prima di avviare la richiesta di certificato "definitivo" ho fatto una prova con questa configurazione, per evitare di mandare richieste a raffica e vedermi il dominio messo in blacklist:
```
#  When your config is ok comment block below to disable use of AMCE staging server
{
    acme_ca https://acme-staging-v02.api.letsencrypt.org/directory 
}

https://labahouse.it { 
  respond "Saluti laba"
  file_server
}
```
Sembra esser andato a buon fine, come si vede nei log

```
"level":"info","ts":1648672689.468747,"logger":"tls","msg":"served key authentication certificate","server_name":"[labahouse.it](http://labahouse.it/)","challenge":"tls-alpn-01","remote":"[3.67.34.92:28868](http://3.67.34.92:28868/)","distributed":false}
{"level":"info","ts":1648672689.7595456,"logger":"tls","msg":"served key authentication certificate","server_name":"[labahouse.it](http://labahouse.it/)","challenge":"tls-alpn-01","remote":"[66.133.109.36:55632](http://66.133.109.36:55632/)","distributed":false}
{"level":"info","ts":1648672690.1461914,"logger":"tls","msg":"served key authentication certificate","server_name":"[labahouse.it](http://labahouse.it/)","challenge":"tls-alpn-01","remote":"[18.236.228.243:21004](http://18.236.228.243:21004/)","distributed":false}
{"level":"info","ts":1648672690.6353579,"logger":"tls.issuance.acme.acme_client","msg":"validations succeeded; finalizing order","order":"https://acme-staging-v02.api.letsencrypt.org/acme/order/49093988/2169015708"}
{"level":"info","ts":1648672691.182047,"logger":"tls.issuance.acme.acme_client","msg":"successfully downloaded available certificate chains","count":1,"first_url":"https://acme-staging-v02.api.letsencrypt.org/acme/ce
rt/fa156c41431753ba62cbcb72220b286c6822"}
{"level":"info","ts":1648672691.1861315,"logger":"tls.obtain","msg":"**certificate obtained successfully**","identifier":"[labahouse.it](http://labahouse.it/)"}
{"level":"info","ts":1648672691.1862698,"logger":"tls.obtain","msg":"releasing lock","identifier":"[labahouse.it](http://labahouse.it/)"}
{"level":"warn","ts":1648672691.429801,"logger":"tls","msg":"stapling OCSP","error":"no OCSP stapling for [[labahouse.it](http://labahouse.it/)]: parsing OCSP response: ocsp: error from server: unauthorized"}
```
c'è solo l'ultima riga che mi lascia qualche perplessità: se ho ben capito  un warning, e non un errore, e quindi non pregiudica le operazioni.
Confermate? posso procedere con la prova e vedere se finalmente riesco a far funzionare i rendirizzamenti su mv.labahouse.it eccetera?
Grazie!

**Reply by snakuzzo** [reply]
Non dovrebbe essere un problema....
Comunque ti basta un attimo per provare...ti colleghi e vedi che ti dà. Con il certificato ottenuto puntando allo staging ti darà sempre un warning sul certificato (ovviamente), ma se guardi le info del certificato vedrai subito se è andato bene oppure no.
A questo punto commenti le prima righe del Caddyfile, riavvii il container e dovrebbe tirare giù i certificato firmato senza problemi.

**Reply by labattazza** [reply]
ciao, grazie mille per il suggerimento.
Alla fine era tutto in ordine, e ora funziona tutto usando i domini di terzo livello: FINALMENTE!!!!

Mi rimane da capire come risolvere un problema di autenticazione del modulo di sincronizzazione di nextcloud, installato sul fisso: inserisco l'url del server nextcloud (sia com ip della rete locale, che come url esterno, quindi https://nuc.labahouse.it/cloud) mi si apre la pagina per l'autenticazione e quindi quella successiva nella quale premendo il pulsante "concedi l'accesso" dovrebbe finalizzare e autenticare il token ma.. non succede nulla.
Non so se voi lo usate e vi è capitato qualcosa di simile.
Intanto il grosso è fatto, grazie alle guide del sito e alle vostre dritte.

**Reply by snakuzzo** [reply]
Non lo uso...ma dovresti vedere sulla console del browser cosa fa

**Reply by labattazza** [reply]
Se può servire a qualcuno, sono riuscito a risolvere. 
Ho dovuto aggiungere le direttive 'overwrite.cli.url', 'overwrite.protocol' e 'overwrite.hostname'.

Questa è la parte di file interessata:
```
<?php
$CONFIG = array (
  'instanceid' => 'XXXXXXXXX',
  'passwordsalt' => 'YYYYYYYYYYYYYYYYYYYY',
  'secret' => 'RANDOM',
  'trusted_proxies' =>
  array (
    0 => '10.0.0.1',
  ),
  'trusted_domains' =>
  array (
    0 => 'labahouse.ddns.net',
    1 => '10.0.0.*',
    2 => '*.labahouse.it',
  ),
  'datadirectory' => '/media/DIRECTORY',
  'dbtype' => 'mysql',
  'version' => '23.0.0.10',
  'overwrite.cli.url' => 'http://10.0.0.5',
  'overwriteprotocol' => 'https',
  'overwritehost' => 'nuc.labahouse.it',
```
